### ISSUE TITLE
Can't run or package "Sandstorm Showcase" from "Packaging tutorial"

### ISSUE THREAD

**Issue by sharpedavid** [sandstorm-io/sandstorm#2797]
I'm following https://docs.sandstorm.io/en/latest/vagrant-spk/packaging-tutorial/ on Windows 7. Everything has gone smoothly to a point.

### Part I: Running the App is DEV mode

I'm on the `vagrant-spk dev` step. In the Windows command prompt I have:

```
C:\Users\David.Sharpe\.babun\cygwin\home\david.sharpe\php-app-to-package-for-sandstorm>vagrant-spk dev
Calling 'vagrant' 'ssh' '-c' '/opt/app/.sandstorm/build.sh && cd /opt/app/.sandstorm && spk dev --pkg-def=/opt/app/.sandstorm/sandstorm-pkgdef.capnp:pkgdef' in C:\Users\David.Sharp
e\.babun\cygwin\home\david.sharpe\php-app-to-package-for-sandstorm\.sandstorm
App is now available from Sandstorm server. Ctrl+C to disconnect.
```

Which is expected. In the Sandstorm admin console at http://local.sandstorm.io:6080/apps I am able to install the "Sandstorm Showcase" app. But when I attempt to run it I get the status circle spinning indefinitely. I checked the "Debug log" and saw the following:

```
Debug log: Untitled Sandstorm Showcase showcase
...** SANDSTORM SUPERVISOR: Starting up grain. Sandbox type: userns
Installing MySQL system tables...
161206  0:25:04 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
161206  0:25:04 [Note] Ignoring --secure-file-priv value as server is running with --bootstrap.
161206  0:25:04 [Note] /usr/sbin/mysqld (mysqld 5.5.53-0+deb8u1) starting as process 31 ...
OK
Filling help tables...
161206  0:25:04 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
161206  0:25:04 [Note] Ignoring --secure-file-priv value as server is running with --bootstrap.
161206  0:25:04 [Note] /usr/sbin/mysqld (mysqld 5.5.53-0+deb8u1) starting as process 37 ...
OK

To start mysqld at boot time you have to copy
support-files/mysql.server to the right place for your system

PLEASE REMEMBER TO SET A PASSWORD FOR THE MySQL root USER !
To do so, start the server, then issue the following commands:

/usr/bin/mysqladmin -u root password 'new-password'
/usr/bin/mysqladmin -u root -h sandbox password 'new-password'

Alternatively you can run:
/usr/bin/mysql_secure_installation

which will also give you the option of removing the test
databases and anonymous user created by default.  This is
strongly recommended for production servers.

See the manual for more instructions.

You can start the MySQL daemon with:
cd /usr ; /usr/bin/mysqld_safe &

You can test the MySQL daemon with mysql-test-run.pl
cd /usr/mysql-test ; perl mysql-test-run.pl

Please report any problems at http://bugs.mysql.com/

waiting for mysql to be available at /var/run/mysqld/mysqld.sock
161206  0:25:05 [Warning] Using unique option prefix key_buffer instead of key_buffer_size is deprecated and will be removed in a future release. Please use the full name instead.
/usr/sbin/mysqld: Error on realpath() on '/var/lib/mysql-files' (Error 2)
161206  0:25:05 [ERROR] Failed to access directory for --secure-file-priv. Please make sure that directory exists and is accessible by MySQL Server. Supplied value : /var/lib/mysql-files
161206  0:25:05 [ERROR] Aborting

waiting for mysql to be available at /var/run/mysqld/mysqld.sock
waiting for mysql to be available at /var/run/mysqld/mysqld.sock
waiting for mysql to be available at /var/run/mysqld/mysqld.sock
```

**The app does not appear to be working. I never see a webpage.**

I stop the `vagrant-spk dev` server using `Ctrl-C` and try the next step. 

### Part II: Packaging the App

I run `vagrant-spk pack ~/projects/package.spk`, and the following message appears on the command line:

```
C:\Users\David.Sharpe\.babun\cygwin\home\david.sharpe\php-app-to-package-for-sandstorm>vagrant-spk pack ~/projects/package.spk
~/projects/package.spk
Calling 'vagrant' 'ssh' '-c' 'cd /opt/app/.sandstorm/ && spk pack --keyring=/host-dot-sandstorm/sandstorm-keyring --pkg-def=/opt/app/.sandstorm/sandstorm-pkgdef.capnp:pkgdef /home/
vagrant/sandstorm-package.spk && spk verify --details /home/vagrant/sandstorm-package.spk && mv /home/vagrant/sandstorm-package.spk /opt/app/sandstorm-package.spk' in C:\Users\Davi
d.Sharpe\.babun\cygwin\home\david.sharpe\php-app-to-package-for-sandstorm\.sandstorm
cpqju2au3nym2825sqnvjs87hn8tfzgygj8383z5gy43cmaqdgn0
{ "appId": "cpqju2au3nym2825sqnvjs87hn8tfzgygj8383z5gy43cmaqdgn0",
  "packageId": "281161cd95556b3bcabdcab0200a1470",
  "title": {"defaultText": "Sandstorm Showcase"},
  "version": 0,
  "marketingVersion": {"defaultText": "0.0.0"},
  "metadata": {
    "icons": {},
    "website": "http:\/\/example.com",
    "codeUrl": "http:\/\/example.com",
    "license": {"none": null},
    "categories": [],
    "author": {"contactEmail": "youremail@example.com", "upstreamAuthor": "Example App Team"},
    "shortDescription": {"defaultText": "one-to-three words"},
    "screenshots": [] } }
Connection to 127.0.0.1 closed.
Traceback (most recent call last):
  File "<string>", line 1028, in <module>
  File "<string>", line 1025, in main
  File "<string>", line 827, in pack
WindowsError: [Error 3] The system cannot find the path specified
vagrant-spk returned -1

C:\Users\David.Sharpe\.babun\cygwin\home\david.sharpe\php-app-to-package-for-sandstorm>
```

The tutorial claims that a file named "package.spk" will be created in the parent directory. I do not see that file, but I do see `sandstorm-package.spk` which has a timestamp to show that it was just updated, so this appears to be the file.

**So the package was created, but this error message sounds significant:** "WindowsError: [Error 3] The system cannot find the path specified".

I tried uploading and installing this file at the Oasis server (https://oasis.sandstorm.io/apps) and achieved the same error messages in the "Debug log".

**I was able to install the TextEditor app to the DEV server, so maybe the issue is limited to the "Showcase" app.**

Installed software:

* Windows 7
* Vagrant 1.9.0
* `vagrant-spk` v0.186
* VirtualBox 5.0.30
* `cmd.exe` (above paths are under Cygwin directory, but I am using `cmd`)

**Comment by sharpedavid** [comment]
I'm surprised that there are no replies yet. Is the report unclear? Is Windows maybe not a well-supported platform for Sandstorm development yet?

**Comment by dwrensha** [comment]
It looks like this is perhaps related to https://github.com/sandstorm-io/vagrant-spk/issues/195.

**Comment by sharpedavid** [comment]
Thanks, dwrensha, I'll follow up on that when time permits.

**Comment by sharpedavid** [comment]
Thanks, @dwrensha. The fix suggested in sandstorm-io/vagrant-spk#195 resolved this for me.

(For posterity: you need to run `vagrant-spk vm provision` to re-execute the `setup.sh` script.)
